Decision relevant metrics

acordonez opened the pull request.

acordonez commented:
The scripts/iqr.py script relies on code in https://github.com/PCMDI/pcmdi_metrics/pull/1235 that I have not pulled into this branch yet.

acordonez commented:
@lee1043 FYI I tested this with numpy version 2.1.3 and did not run into any issues with xclim.

acordonez commented:
I've compared the following metrics with Tempest Extremes outputs and they match within 1e-5 or 1e-6 with the differences following a noise pattern using the latest code changes.
ANN frequency:
mean precip
mean tasmax
txx
mean tasmin
mean tasmin < 32F

DJF, MAM, JJA, and SON frequency:
mean precip

Larger differences were seen in these quantities. Some of these have been investigated (days > 100 for example is very sensitive). It's possible the quantiles need to be looked at further, or maybe this level of difference is also expected.
precip q50 ANN (-0.2 to 0.3)
tasmax > 100F (average dif ~30)
tasmax 99.9 (differences between 0-0.1)

Any metrics not listed here were not investigated with the latest code. However they had been previously compared with Tempest Extremes (except continuous wet days and continuous dry days) and for the non-quantile and non-threshold metrics there were not large differences seen.

acordonez commented:
I've added some run scripts to the param/ directory that users can reference.

acordonez commented:
@lee1043 I'm not sure why this is failing the pre-commit check. All my files are passing pre-commit when I run it locally. Using pre-commit 4.1.0 and installed PMP from this branch.

lee1043 commented:
@acordonez it looks like a few files under `drcdm/param/*` were missed for pre-commit. I added my commit to the branch to fix them, and pre-commit CI is passing now. I was thinking of opening a new branch from your branch for my change, to not mess up your code, but accidently added my commits to your branch directly -- sorry for that. You might want to to `git pull` first before making any further changes.

acordonez commented:
@lee1043 Not a problem, thanks for finding that. It's so weird that they don't show up as missing for me!

acordonez commented:
@lee1043  I think this is as good as it's going to get on my end. I've added some notes in the comments to this PR and added a note to the README pointing back here. I don't have a demo but you could probably test run this on NERSC using the parameter files in param/PRISM with a few minor changes.

acordonez commented:
@jsg5776 adding you as an FYI. I'm wrapping up the small set of metrics I worked on initially. I've also added a bunch of parameter files and scripts for running on NERSC over downscaled data.

lee1043 commented:
@acordonez could you point me to the steps or instructions if any on how I can test this code?

lee1043 commented:
@acordonez Thank you for the info. The code started running on NERSC for PRISM, and it looks like it will take a while. How long does it usually take?

lee1043 commented:
It looks like the job got killed maybe because of wall clock limit, when I just ran it on the login node. Do you usually submit the job to the computing node?

```
2025-03-06 00:28:18,392 [INFO]: font_manager.py(_load_fontmanager:1639) >> generated new fontManager
2025-03-06 00:28:18,392 [INFO]: font_manager.py(_load_fontmanager:1639) >> generated new fontManager
2025-03-06 00:28:19,609 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining 'percent' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 00:28:19,609 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining 'percent' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 00:28:19,610 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining '%' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 00:28:19,610 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining '%' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 00:28:19,610 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining 'year' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 00:28:19,610 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining 'year' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 00:28:19,610 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining 'yr' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 00:28:19,610 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining 'yr' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 00:28:19,610 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining 'C' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 00:28:19,610 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining 'C' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 00:28:19,611 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining 'd' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 00:28:19,611 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining 'd' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 00:28:19,611 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining 'h' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 00:28:19,611 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining 'h' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 00:28:19,611 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining 'degrees_north' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 00:28:19,611 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining 'degrees_north' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 00:28:19,612 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining 'degrees_east' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 00:28:19,612 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining 'degrees_east' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 00:28:19,612 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining 'degrees' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 00:28:19,612 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining 'degrees' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 00:28:19,612 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining '[speed]' (<class 'pint.delegates.txt_defparser.plain.DerivedDimensionDefinition'>)
2025-03-06 00:28:19,612 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining '[speed]' (<class 'pint.delegates.txt_defparser.plain.DerivedDimensionDefinition'>)

Metrics output path not found.
Creating metrics output directory /pscratch/sd/l/lee1043/PMP/drcdm_test/output/obs/pr/PRISM/
No sftlf file found for Reference None

-----------------------
model, run, variable: Reference None pr
test_data (model in this case) full_path:
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_19810101-19811231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_19820101-19821231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_19830101-19831231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_19840101-19841231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_19850101-19851231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_19860101-19861231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_19870101-19871231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_19880101-19881231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_19890101-19891231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_19900101-19901231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_19910101-19911231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_19920101-19921231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_19930101-19931231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_19940101-19941231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_19950101-19951231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_19960101-19961231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_19970101-19971231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_19980101-19981231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_19990101-19991231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_20000101-20001231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_20010101-20011231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_20020101-20021231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_20030101-20031231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_20040101-20041231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_20050101-20051231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_20060101-20061231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_20070101-20071231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_20080101-20081231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_20090101-20091231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_20100101-20101231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_20110101-20111231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_20120101-20121231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_20130101-20131231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_20140101-20141231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_20150101-20151231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_20160101-20161231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_20170101-20171231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_20180101-20181231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_20190101-20191231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_20200101-20201231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_20210101-20211231.nc
   /pscratch/sd/l/lee1043/obs4MIPs/OSU/PRISM-M3/day/pr/gn/latest/pr_day_PRISM-M3_PCMDI_gn_20220101-20220930.nc
Generating land sea mask.
Metric: annualmean_pr
Metric seasonalmean_pr
Metric: annual_pxx
Metric: wettest_5yr
Metric: annual_cwd
2025-03-06 01:23:04,290 [WARNING]: bounds.py(create_bounds:987) >> The 'lat' coordinate variable is missing a 'units' attribute. Assuming 'units' is 'degrees_north'.
2025-03-06 01:23:04,290 [WARNING]: bounds.py(create_bounds:987) >> The 'lat' coordinate variable is missing a 'units' attribute. Assuming 'units' is 'degrees_north'.
./run.sh: line 7: 216963 Killed                  drcdm_driver.py -p your_parameter_file.py
```

acordonez commented:
@lee1043  Yes I have to run this on the computing node at NERSC. I think the temperature variables only take ~1.5 hours now and precip is a little longer. I have example jobs scripts in the parameter folder, eg https://github.com/PCMDI/pcmdi_metrics/blob/1098_ao_drcdm/pcmdi_metrics/drcdm/param/PRISM/run_CONUS_obs_pr.sh

lee1043 commented:
@acordonez in my test on gates, I got the following error when using ACCESS-CM2 historical simulation. Any chance it look familiar to you?

Part of my parameter file:
```python
test_data_set = ["ACCESS-CM2"]
realization = ["r1i1p1f1"]
test_data_path = "/p/css03/esgf_publish/CMIP6/CMIP/CSIRO-ARCCSS/ACCESS-CM2/historical/r1i1p1f1/day/pr/gn/v20191108/"
filename_template = "%(variable)_day_ACCESS-CM2_historical_r1i1p1f1_gn_*.nc"
```

Error log:
```
2025-03-06 14:29:08,055 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining 'percent' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 14:29:08,055 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining 'percent' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 14:29:08,055 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining '%' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 14:29:08,055 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining '%' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 14:29:08,055 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining 'year' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 14:29:08,055 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining 'year' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 14:29:08,055 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining 'yr' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 14:29:08,055 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining 'yr' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 14:29:08,056 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining 'C' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 14:29:08,056 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining 'C' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 14:29:08,056 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining 'd' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 14:29:08,056 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining 'd' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 14:29:08,056 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining 'h' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 14:29:08,056 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining 'h' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 14:29:08,057 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining 'degrees_north' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 14:29:08,057 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining 'degrees_north' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 14:29:08,057 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining 'degrees_east' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 14:29:08,057 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining 'degrees_east' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 14:29:08,057 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining 'degrees' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 14:29:08,057 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining 'degrees' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
2025-03-06 14:29:08,057 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining '[speed]' (<class 'pint.delegates.txt_defparser.plain.DerivedDimensionDefinition'>)
2025-03-06 14:29:08,057 [WARNING]: registry.py(_helper_single_adder:541) >> Redefining '[speed]' (<class 'pint.delegates.txt_defparser.plain.DerivedDimensionDefinition'>)
No sftlf file found for ACCESS-CM2 r1i1p1f1

-----------------------
model, run, variable: ACCESS-CM2 r1i1p1f1 pr
test_data (model in this case) full_path:
   /p/css03/esgf_publish/CMIP6/CMIP/CSIRO-ARCCSS/ACCESS-CM2/historical/r1i1p1f1/day/pr/gn/v20191108/pr_day_ACCESS-CM2_historical_r1i1p1f1_gn_18500101-18991231.nc
   /p/css03/esgf_publish/CMIP6/CMIP/CSIRO-ARCCSS/ACCESS-CM2/historical/r1i1p1f1/day/pr/gn/v20191108/pr_day_ACCESS-CM2_historical_r1i1p1f1_gn_19000101-19491231.nc
   /p/css03/esgf_publish/CMIP6/CMIP/CSIRO-ARCCSS/ACCESS-CM2/historical/r1i1p1f1/day/pr/gn/v20191108/pr_day_ACCESS-CM2_historical_r1i1p1f1_gn_19500101-19991231.nc
   /p/css03/esgf_publish/CMIP6/CMIP/CSIRO-ARCCSS/ACCESS-CM2/historical/r1i1p1f1/day/pr/gn/v20191108/pr_day_ACCESS-CM2_historical_r1i1p1f1_gn_20000101-20141231.nc
Generating land sea mask.
Metric: annualmean_pr
Traceback (most recent call last):
  File "/home/lee1043/.conda/envs/pcmdi_metrics_dev_20241216/bin/drcdm_driver.py", line 411, in <module>
    result_dict = compute_metrics.get_annualmean_pr(
  File "/home/lee1043/.conda/envs/pcmdi_metrics_dev_20241216/lib/python3.10/site-packages/pcmdi_metrics/drcdm/lib/compute_metrics.py", line 881, in get_annualmean_pr
    PR = TimeSeriesData(ds, varname)
  File "/home/lee1043/.conda/envs/pcmdi_metrics_dev_20241216/lib/python3.10/site-packages/pcmdi_metrics/drcdm/lib/compute_metrics.py", line 22, in __init__
    self.freq = xr.infer_freq(ds.time)
  File "/home/lee1043/.conda/envs/pcmdi_metrics_dev_20241216/lib/python3.10/site-packages/xarray/coding/frequencies.py", line 89, in infer_freq
    raise ValueError("'index' must contain datetime-like objects")
ValueError: 'index' must contain datetime-like objects
```

lee1043 commented:
Regarding the test on gates after adjusting `msyear` and `meyear`, it looks like it runs without any issues! 

```
model, run, variable: ACCESS-CM2 r1i1p1f1 pr
test_data (model in this case) full_path:
   /p/css03/esgf_publish/CMIP6/CMIP/CSIRO-ARCCSS/ACCESS-CM2/historical/r1i1p1f1/day/pr/gn/v20191108/pr_day_ACCESS-CM2_historical_r1i1p1f1_gn_18500101-18991231.nc
   /p/css03/esgf_publish/CMIP6/CMIP/CSIRO-ARCCSS/ACCESS-CM2/historical/r1i1p1f1/day/pr/gn/v20191108/pr_day_ACCESS-CM2_historical_r1i1p1f1_gn_19000101-19491231.nc
   /p/css03/esgf_publish/CMIP6/CMIP/CSIRO-ARCCSS/ACCESS-CM2/historical/r1i1p1f1/day/pr/gn/v20191108/pr_day_ACCESS-CM2_historical_r1i1p1f1_gn_19500101-19991231.nc
   /p/css03/esgf_publish/CMIP6/CMIP/CSIRO-ARCCSS/ACCESS-CM2/historical/r1i1p1f1/day/pr/gn/v20191108/pr_day_ACCESS-CM2_historical_r1i1p1f1_gn_20000101-20141231.nc
Generating land sea mask.
Metric: annualmean_pr
Qt: Session management error: None of the authentication protocols specified are supported
Metric seasonalmean_pr
Metric: annual_pxx
Metric: wettest_5yr
Metric: annual_cwd
2025-03-06 15:12:32,139 [WARNING]: bounds.py(create_bounds:987) >> The 'lat' coordinate variable is missing a 'units' attribute. Assuming 'units' is 'degrees_north'.
2025-03-06 15:12:32,139 [WARNING]: bounds.py(create_bounds:987) >> The 'lat' coordinate variable is missing a 'units' attribute. Assuming 'units' is 'degrees_north'.
/home/lee1043/.conda/envs/pcmdi_metrics_dev_20241216/lib/python3.10/site-packages/pcmdi_metrics/drcdm/lib/compute_metrics.py:1218: UserWarning: Variable time has datetime type and a bounds variable but time.encoding does not have units specified. The units encodings for time and time_bnds will be determined independently and may not be equal, counter to CF-conventions. If this is a concern, specify a units encoding for time before writing to a file.
  Pcwd.to_netcdf(nc_file, "w")
Metric: annual_cdd
2025-03-06 15:13:00,748 [WARNING]: bounds.py(create_bounds:987) >> The 'lat' coordinate variable is missing a 'units' attribute. Assuming 'units' is 'degrees_north'.
2025-03-06 15:13:00,748 [WARNING]: bounds.py(create_bounds:987) >> The 'lat' coordinate variable is missing a 'units' attribute. Assuming 'units' is 'degrees_north'.
/home/lee1043/.conda/envs/pcmdi_metrics_dev_20241216/lib/python3.10/site-packages/pcmdi_metrics/drcdm/lib/compute_metrics.py:1255: UserWarning: Variable time has datetime type and a bounds variable but time.encoding does not have units specified. The units encodings for time and time_bnds will be determined independently and may not be equal, counter to CF-conventions. If this is a concern, specify a units encoding for time before writing to a file.
  Pcdd.to_netcdf(nc_file, "w")
Metric: pr_q50
Metric: pr_q99p9
Metric: pr_q99p0
INFO::2025-03-06 15:15::pcmdi_metrics:: Results saved to a json file: /p/user_pub/climate_work/lee1043/cdat/pmp/drcdm/test_output/ACCESS-CM2_metrics.json
2025-03-06 15:15:05,314 [INFO]: base.py(write:422) >> Results saved to a json file: /p/user_pub/climate_work/lee1043/cdat/pmp/drcdm/test_output/ACCESS-CM2_metrics.json
2025-03-06 15:15:05,314 [INFO]: base.py(write:422) >> Results saved to a json file: /p/user_pub/climate_work/lee1043/cdat/pmp/drcdm/test_output/ACCESS-CM2_metrics.json
INFO::2025-03-06 15:15::pcmdi_metrics:: Results saved to a json file: /p/user_pub/climate_work/lee1043/cdat/pmp/drcdm/test_output/decision_relevant_metrics.json
2025-03-06 15:15:22,485 [INFO]: base.py(write:422) >> Results saved to a json file: /p/user_pub/climate_work/lee1043/cdat/pmp/drcdm/test_output/decision_relevant_metrics.json
2025-03-06 15:15:22,485 [INFO]: base.py(write:422) >> Results saved to a json file: /p/user_pub/climate_work/lee1043/cdat/pmp/drcdm/test_output/decision_relevant_metrics.json
```

lee1043 commented:
@acordonez thank you for this important contribution!

acordonez commented:
@lee1043 pointed out that the quantile metrics are not currently being saved to the JSON file. This is a known issue and some work needs to be done to get the metrics calculation code to work on quantities without a time axis.
